api/v1alpha1/gitrepository_types.go

```diff
@@ -0,0 +1,75 @@
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type GitRepositorySpec struct {
+	Source GitRepositorySource `json:"source,omitempty"`
+	// GitURL is the base URL of Git server used for API calls.
+	// +kubebuilder:validation:Required
+	// +kubebuilder:validation:Pattern=`^https?:\/\/.+$`
+	GitURL string `json:"gitURL"`
+	// SecretRef is the reference to secret that contain Git server credentials
+	// +kubebuilder:validation:Optional
+	SecretRef SecretReference `json:"secretRef"`
+}
+
+type GitRepositorySource struct {
+	// +kubebuilder:validation:Enum:=argocd;backstage;crossplane;gitea;nginx
+	// +kubebuilder:validation:Optional
+	EmbeddedAppName string `json:"embeddedAppName"`
+	// Path is the absolute path to directory that contains Kustomize structure or raw manifests.
+	// This is required when Type is set to local.
+	// +kubebuilder:validation:Optional
+	Path string `json:"path"`
+	// Type is the source type.
+	// +kubebuilder:validation:Enum:=local;embedded
+	// +kubebuilder:default:=embedded
+	Type string `json:"type"`
```

Comment:
Can you describe what is the purpose of the type and what will happen if we select one the values: `local`, `embedded`, etc  ?

Reply:
Local indicates the gitea repository contents will come from a local directory. E.g. `/home/ubuntu/myrepo`

Embedded indicates the repository contents will come from what is embedded in the go binary.